tcms/testcases/tests/test_models.py

```diff
@@ -162,6 +162,36 @@ def test_send_mail_to_case_author(self, send_mail):
         )
 
 
+class TestCaseCalculateExpectedDuration(TestCase):
+    """Test TestCase.expected_duration"""
+    
+    @parameterized.expand(
+        [
+            (
+                "both_values_are_set",
+                timezone.datetime(2021, 3, 22),
+                timezone.datetime(2021, 3, 23),
+                timezone.timedelta(days=1),
+            ),
+            ("both_values_are_none", None, None, None),
+
+            ("setup_duration_is_none", None, timezone.datetime(2021, 3, 23), None),
+            ("testing_duration_is_none", timezone.datetime(2021, 3, 22), None, None),
+        ]
+    )
+    
+    def test_expected_duration_when(
+            self, _name, setup_duration, testing_duration, expected
+    ):
+        execution = TestExecutionFactory(
+            setup_duration=setup_duration, testing_duration=testing_duration
+        )
+        self.assertEqual(execution.expected_duration, expected)
+            ("start_date_is_none", None, timezone.datetime(2021, 3, 23), None),
+            ("stop_date_is_none", timezone.datetime(2021, 3, 22), None, None),
+        ]
+    )
```

Comment:
This test scenario is a verbatim copy of lines 195-212 below, it's not going to work.  You need to test the new property in the TestCase model class!